### PR TITLE
Fixes issue with shiny and uploading files.

### PR DESCRIPTION
**Pull request by karthiknadig**
#3283

### CODE REVIEW

**Comment by MikhailArkhipov** [comment]
👍 . @int19h  - need 2nd.

**Comment by MikhailArkhipov** [comment]
Or @AlexanderSher please

File: src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs [hunk 1]
```diff
@@ -55,7 +55,7 @@ public class RemoteUriWebService : IRemoteUriWebService {
             var remoteUri = GetRemoteUri(context.Request.Url, remoteBaseUrl);
             request.Headers.Add(CustomHttpHeaders.RTVSRequestedURL, remoteUri.ToString());
 
-            if (context.Request.InputStream.CanSeek && context.Request.InputStream.Length > 0) {
+            if (context.Request.ContentLength64 > 0) {
```
Comment (AlexanderSher) [inline src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs]: Why we don't need to check for `CanSeek` anymore?

Comment (karthiknadig) [inline src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs]: Seek was needed to query the length from a stream for POST messages. Using the stream directly to determine the length requires additional conditions. For, example, when a file is uploaded, the stream is *not* seek-able but read-only. For post messages, the stream may be seek-able and length is only available when it is seek-able. `ContentLength64` stores the message body length for all request objects, and it is simpler. The issue here was caused by read-only case I mentioned above.

Comment (AlexanderSher) [inline src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs]: So `ContentLength64 > 0` only if we can use the stream, correct?

Comment (karthiknadig) [inline src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs]: `ContentLength64 > 0` only if the stream has content. Also, we need to do the copy only if the stream has content. The stream is still usable in the websocket context.

Comment (karthiknadig) [inline src/Host/Client/Impl/BrokerServices/RemoteUriWebService.cs]: Opening the stream when there is not supposed to be content will lead to error. So, the request stream has to be accessed only if there is content or in websocket context.

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**